lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -288,14 +288,7 @@ SmallVector<unsigned> getOrder(Attribute layout) {
     auto distributedLayout = cast<DistributedEncodingTrait>(layout);
     auto rank = distributedLayout.getWarpsPerCTA().size();
     SmallVector<unsigned> order(rank);
-    auto nvidiaMma = dyn_cast<NvidiaMmaEncodingAttr>(layout);
-    if (nvidiaMma && nvidiaMma.isHopper()) {
-      // Hopper WGMMA requires that the 4 warps in a warp group be laid out along
-      // the M-dimension; so here we rasterize M-first
-      std::iota(order.begin(), order.end(), 0);
-    } else {
-      std::iota(order.rbegin(), order.rend(), 0);
-    }
+    std::iota(order.rbegin(), order.rend(), 0);
```

Comment:
I am a bit confused. We cannot simply change the order here and move on, as that would mean that we are iterating the elements in a different order. Now, when I was going to point at the code to change, I saw that code was already N-major! See
https://github.com/triton-lang/triton/blob/3ca2f498e98ed7249b82722587c511a5610e00c4/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/WGMMA.cpp#L423-L424
so, I think we just never call `getOrder` with an `mmav3`! (at least not in any place that matters).

I then went to check the LL lowering and we have
https://github.com/triton-lang/triton/blob/3ca2f498e98ed7249b82722587c511a5610e00c4/lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp#L319
which means that we are also iterating in the `[1, 0]` order (the order is given by the order of the out dims, we should document this somewhere).

All this to say that, yes, this change LGTM. cc @Jokeren for visibility.

Something we should do sooner than later is put here a 
```c++
if (mma.isHopper())
  llvm::report_fatal_error("Test");
```
run CI and audit all the calls to this function, as they are likely to be incorrect uses of `getOrder`.